x-pack/plugins/security_solution/public/detection_engine/rule_gaps/components/manual_rule_run/index.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiBetaBadge,
+  EuiConfirmModal,
+  EuiDatePicker,
+  EuiDatePickerRange,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiHorizontalRule,
+  EuiSpacer,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import moment from 'moment';
+import React, { useCallback, useState } from 'react';
+import { TECHNICAL_PREVIEW, TECHNICAL_PREVIEW_TOOLTIP } from '../../../../common/translations';
+
+import * as i18n from './translations';
+
+interface ManualRuleRunModalProps {
+  onCancel: () => void;
+  onConfirm: (timeRange: { startDate: moment.Moment; endDate: moment.Moment }) => void;
+}
+
+const ManualRuleRunModalComponent = ({ onCancel, onConfirm }: ManualRuleRunModalProps) => {
+  const modalTitleId = useGeneratedHtmlId();
+
+  const [startDate, setStartDate] = useState(moment().subtract(1, 'd'));
+  const [endDate, setEndDate] = useState(moment());
+  const isInvalid = startDate.isAfter(endDate);
```

Comment:
It looks like we can submit but the API throws an error if `startDate` and `endDate` are equal. Super minor edge case but if it's not too hard, we should match this verification.
<img width="326" alt="Screenshot 2024-06-03 at 12 31 01 PM" src="https://github.com/elastic/kibana/assets/56367316/7eb56af2-010a-486f-9093-84f27723a243">

Reply:
Good point! Fixed